packages/ui/react-ui-deck/src/components/Attention/AttentionProvider.tsx

```diff
@@ -6,22 +6,73 @@
 
 // eslint-disable-next-line unused-imports/no-unused-imports
 import { createContext, type Scope, type CreateScope } from '@radix-ui/react-context';
+import { useControllableState } from '@radix-ui/react-use-controllable-state';
 // eslint-disable-next-line unused-imports/no-unused-imports
-import React from 'react';
+import React, { ComponentProps, type FocusEvent, type HTMLAttributes, type PropsWithChildren, useState } from 'react';
 
 const ATTENTION_NAME = 'Attention';
 
 type AttentionContextValue = {
   attended: Set<string>;
 };
 
-const [AttentionProvider, useAttentionContext] = createContext(ATTENTION_NAME, { attended: new Set() });
+const [AttentionContextProvider, useAttentionContext] = createContext<AttentionContextValue>(ATTENTION_NAME, {
+  attended: new Set(),
+});
 
 const useHasAttention = (attendableId?: string) => {
   const { attended } = useAttentionContext(ATTENTION_NAME);
-  return attended.has(attendableId);
+  return attendableId ? attended.has(attendableId) : false;
 };
 
-export { AttentionProvider, useAttentionContext, useHasAttention, ATTENTION_NAME };
+/**
+ * Computes HTML element attributes to apply so the attention system can detect changes
+ * @param attendableId
+ */
+const useAttendable = (attendableId?: string) => {
+  return { 'data-attendable-id': attendableId };
+};
+
+const getAttendables = (cursor: FocusEvent['target'], acc: string[] = []): string[] => {
+  const closestAttendable = cursor.closest('[data-attendable-id]');
+  if (!closestAttendable) {
+    return acc;
+  } else {
+    acc.push(closestAttendable.getAttribute('data-attendable-id')!);
+    // (attempt tail recursion)
+    return !closestAttendable.parentElement ? acc : getAttendables(closestAttendable.parentElement, acc);
+  }
+};
```

Comment:
Can you add a doc comment explaining the semantics of this? Especially the behaviour of `cursor.closest(['data-attendable-id'])`.

Just so I get this clear in my head, we're searching up the DOM and collecting all of the attendable id's along the way. How can you supply an `Element` in place of a `FocusEvent['target']`?

Reply:
> How can you supply an Element in place of a FocusEvent['target']?

This is demonstrated in line 43, `closestAttendable.parentElement` satisfies the type. Will add a JSDoc to explain the rest, certainly.